core/cache/charm.go

```diff
@@ -48,6 +48,7 @@ func (c *Charm) setDetails(details CharmChange) {
 	})
 
 	c.details = details
+	c.hub.Publish(modelCharmMetadataChanged, details)
```

Comment:
The whole charm is changing, not just the metadata.

```suggestion
	c.hub.Publish(modelCharmChanged, details)
```

Reply:
Ah right. Let me change this.